command_line_assistant/logging.py

```diff
@@ -0,0 +1,48 @@
+#: Logging configuration
+import logging.config
+
+from command_line_assistant.config import Config
+
+LOGGING_CONFIG_DICTIONARY = {
+    "version": 1,
+    "level": "INFO",
+    "formatters": {
+        "verbose": {
+            "format": "[%(asctime)s] [%(filename)s:%(lineno)d] %(levelname)s: %(message)s",
+            "datefmt": "%m/%d/%Y %I:%M:%S %p",
+        },
+        "minimal": {"format": "%(message)s"},
+    },
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+            "formatter": "minimal",
+            "level": "INFO",
+            "stream": "ext://sys.stdout",
+        },
+        "file": {
+            "class": "logging.handlers.RotatingFileHandler",
+            "formatter": "verbose",
+            "filename": "",
+        },
+    },
+    "loggers": {"root": {"handlers": ["file"], "level": "INFO"}},
+}
+
+
+def setup_logging(config: Config, verbose: bool = False):
+    """Setup basic logging functionality"""
+    logging_file = config.logging.file
+
+    if not logging_file.parent.exists():
+        logging_file.parent.mkdir(mode=0o755)
+
+    logging_configuration = LOGGING_CONFIG_DICTIONARY.copy()
```

Comment:
In order to avoid modifying the original dictionary, `copy.deepcopy()` needs to be used.

Reply:
good catch!

Reply:
```suggestion
    logging_configuration = LOGGING_CONFIG_DICTIONARY.deepcopy()
```